cmd/kube-proxy/app/server.go

```diff
@@ -248,7 +248,7 @@ func newProxyServer(ctx context.Context, config *kubeproxyconfig.KubeProxyConfig
 
 	err, fatal := checkBadIPConfig(s, dualStackSupported)
 	if err != nil {
-		if fatal {
+		if fatal || (config.ConfigHardFail != nil && *config.ConfigHardFail) {
```

Comment:
you should update the defaulting (`pkg/proxy/apis/config/*/defaults.go`) to fill this in if it's not set so that you can just assume it's not nil here

Reply:
v1alpha1 won't have `ConfigHardFail`. So if a user comes with v1alpha1 then then config.ConfigHardFail will be always nil, right?
(config being internal config after conversion)